src/components/CodeInput/index.js

```diff
@@ -82,6 +83,8 @@ const CodeInput = React.forwardRef((props, ref) => {
                     <RequiredAsterisk required={required} />
                     {label}
                 </StyledLabel>
+
+                <StyledHiddenLabel id={labelId}>{label}</StyledHiddenLabel>
```

Comment:
Take a look here: https://www.w3.org/WAI/tutorials/forms/custom-controls/
In order to have the right a11y we should have an label for each input but since we don't have a clear understanding of how a11y should be implemented here we should not do this for now

Reply:
sure, definitely, I applied changes from my a11y perspective, might be diffuse, I understand completely... however it was quite a new learning and good exercise.